lib/addons/core-addon/index.ts

```diff
@@ -20,11 +40,25 @@ export class CoreAddOn implements ClusterAddOn {
     }
 
     deploy(clusterInfo: ClusterInfo): void {
+
+        // Create a service account if user provides namespace and service account
+        let serviceAccountRoleArn: string | undefined = undefined;
+        let namespace: string = this.coreAddOnProps?.namespace ? this.coreAddOnProps.namespace : DEFAULT_NAMESPACE;
```

Comment:
Please correct me if I am wrong, but if customers supply a custom namespace name (e.g. ebs-addon) then this code will fail since namespace won't exist. 
If my line of reasoning is correct, then let's use namespace-utils.createNamespace. This one won't fail if it already exists.

Reply:
@shapirov103 : My thinking was that while currently all add-ons go into Kube-system, allowing them to pass it in would future proof the core-addon since it is a building block. If that is not necessary, let me know and I can change it.

Reply:
I think it is a good option to add. Some customers have weird organization of namespaces. Let's just call createNamespace (e.g. kubectl apply through our utils) and we should be good.

Reply:
@shapirov103 looks like `CfnAddon()` accepts `serviceAccountRoleArn` but no `namespace` as parameter. Will  EKS addon honor the `namespace` passed via a `serviceAccount`? Or are we just future proofing here?

Reply:
We don't need to future-proof it in my opinion. It may confuse customers. Plus it is taking serviceAccountRoleArn rather than sa itself so I doubt it will honor the sa namespace (@camba1 have you tested with it?).